Description: Update zip dependency to 2.5.0 for Debian compatibility
 Upstream depends on zip 0.6.3, but Debian provides zip v2. The API
 has changed since 0.6, so this patch updates Cargo.toml and the
 code accordingly
Author: Nadzeya Hutsko <nadzya.info@gmail.com>
Bug: https://github.com/bokuweb/docx-rs/issues/857
Last-Update: 2025-09-19
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -246,7 +246,7 @@
 version = "0.8.4"
 
 [dependencies.zip]
-version = "0.6.3"
+version = "2"
 features = ["deflate"]
 default-features = false
 
--- a/src/zipper/mod.rs
+++ b/src/zipper/mod.rs
@@ -10,12 +10,12 @@
 {
     let mut zip = zip::ZipWriter::new(w);
 
-    zip.add_directory("word/", Default::default())?;
-    zip.add_directory("word/_rels", Default::default())?;
-    zip.add_directory("_rels/", Default::default())?;
-    zip.add_directory("docProps/", Default::default())?;
+    zip.add_directory::<&str, ()>("word/", FileOptions::default())?;
+    zip.add_directory::<&str, ()>("word/_rels", FileOptions::default())?;
+    zip.add_directory::<&str, ()>("_rels/", FileOptions::default())?;
+    zip.add_directory::<&str, ()>("docProps/", FileOptions::default())?;
 
-    let options = FileOptions::default()
+    let options: FileOptions<'_, ()> = FileOptions::default()
         .compression_method(zip::CompressionMethod::Stored)
         .unix_permissions(0o755);
 
@@ -69,7 +69,7 @@
     }
 
     if !xml.media.is_empty() {
-        zip.add_directory("word/media/", Default::default())?;
+        zip.add_directory::<&str, ()>("word/media/", FileOptions::default())?;
         for m in xml.media {
             // For now only png supported
             zip.start_file(format!("word/media/{}.png", m.0), options)?;
@@ -79,11 +79,11 @@
 
     // For now support only taskpanes
     if let Some(taskpanes) = xml.taskpanes {
-        zip.add_directory("word/webextensions/", Default::default())?;
+        zip.add_directory::<&str, ()>("word/webextensions/", FileOptions::default())?;
         zip.start_file("word/webextensions/taskpanes.xml", options)?;
         zip.write_all(&taskpanes)?;
 
-        zip.add_directory("word/webextensions/_rels", Default::default())?;
+        zip.add_directory::<&str, ()>("word/webextensions/_rels", FileOptions::default())?;
         zip.start_file("word/webextensions/_rels/taskpanes.xml.rels", options)?;
         zip.write_all(&xml.taskpanes_rels)?;
 
@@ -97,7 +97,7 @@
     }
 
     if !xml.custom_items.is_empty() {
-        zip.add_directory("customXml/_rels", Default::default())?;
+        zip.add_directory::<&str, ()>("customXml/_rels", FileOptions::default())?;
     }
 
     for (i, item) in xml.custom_items.into_iter().enumerate() {
